packages/ra-test/src/TestContext.tsx

```diff
@@ -63,7 +63,19 @@ export const TestContext = (props: TestContextProps) => {
         enableReducers = false,
         customReducers = {},
     } = props;
-    const history = useRef<History>(props.history || createMemoryHistory());
+    let historyRef = React.useRef<History>();
+    if (historyRef.current == null) {
+        historyRef.current = createMemoryHistory();
+    }
+
+    let finalHistory = historyRef.current;
+    let [historyState, setHistoryState] = React.useState({
```

Comment:
You're doing the same strange stuff as in CoreAdminContext. Why not reuse that component?

Reply:
Because I believe we're going to remove `ra-test`